.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Test
+
+on:
+  [push, pull_request] # Pull request trigger should run the tests on merged files
+   
+# Not currently using a build type
+# env:
+#   # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+#   BUILD_TYPE: Release
+
+jobs:
+  build:
+    # This script uses linux specific commands, so is unlikely to work on a windows runner.
+    runs-on: self-hosted
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      working-directory: ${{github.workspace}}
+      run: setup.sh
```

Comment:
This script contains "echo "To build all and run tests, do the following: $ cd build && make && cd unit_tests && ctest"" Which is a little out of place for the CI. Is it bad practice to have it shouting that in to the void?

Reply:
is it shouting into the void? aren't the logs stored?

Reply:
It is storing it as a log file, but you can't exactly run the command from the log file. I'm mostly just wondering if that's bad practice or we'll put up with it.

Reply:
I don't see a problem with it, imo it's ideal to have the CI using scripts as close as possible to what an end user would be. That means that if our scripts start breaking for some reason we will know asap.

If you are really bothered you could add `> /dev/null` to send stout to null